File: claude/.claude/skills/shortcut/scripts/workflows.py
#!/usr/bin/env python3
"""Shortcut Workflows - Operations for workflow management."""

import sys
import json
import argparse
from typing import Dict, List
from shortcut_client import ShortcutClient, format_workflow


def get_workflow(workflow_id: int) -> Dict:
    """Get a single workflow by ID."""
    client = ShortcutClient()
    workflow = client.get(f"workflows/{workflow_id}")
    return format_workflow(workflow)


def list_workflows() -> List[Dict]:
    """List all workflows in the workspace."""
    client = ShortcutClient()
    workflows = client.get("workflows")
    return [format_workflow(w) for w in workflows]


def main():
    parser = argparse.ArgumentParser(description="Shortcut Workflows Operations")
    subparsers = parser.add_subparsers(dest="command")

    get_parser = subparsers.add_parser("get")
    get_parser.add_argument("workflow_id", type=int)

    subparsers.add_parser("list")

    args = parser.parse_args()

    try:
        if args.command == "get":
            result = get_workflow(args.workflow_id)
        elif args.command == "list":
            result = list_workflows()
        else:
            parser.print_help()
            sys.exit(1)

        print(json.dumps(result, indent=2))
    except Exception as e:
        print(json.dumps({"error": str(e)}, indent=2), file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()

File: claude/.claude/skills/shortcut/src/entities/workflows.ts
import type { Command } from "commander";
import type { ApiWorkflow, FormattedWorkflow } from "../types.js";
import {
  ShortcutClient,
  formatWorkflow,
  output,
  outputError,
} from "../client.js";

export async function getWorkflow(
  workflowId: number,
): Promise<FormattedWorkflow> {
  const client = new ShortcutClient();
  const workflow = await client.get<ApiWorkflow>(`workflows/${workflowId}`);
  return formatWorkflow(workflow);
}

export async function listWorkflows(): Promise<FormattedWorkflow[]> {
  const client = new ShortcutClient();
  const workflows = await client.get<ApiWorkflow[]>("workflows");
  return workflows.map(formatWorkflow);
}

export function registerReadCommands(program: Command): void {
  const workflows = program
    .command("workflows")
    .description("Workflow operations");

  workflows
    .command("get <workflow-id>")
    .description("Get a workflow by ID")
    .action(async (workflowId: string) => {
      try {
        const result = await getWorkflow(parseInt(workflowId, 10));
        output(result);
      } catch (e) {
        outputError(e);
      }
    });

  workflows
    .command("list")
    .description("List all workflows")
    .action(async () => {
      try {
        const result = await listWorkflows();
        output(result);
      } catch (e) {
        outputError(e);
      }
    });
}
